### PR TITLE
virt-handler: Refactor claimDeviceOwnership method

### PR DESCRIPTION
**Pull request by EdDev**
**What this PR does / why we need it**:

Pass to the method the isolation data it needs to avoid re-calculating
it multiple times in the setup flow.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by ormergi** [comment]
Note that with this PR changes the calling order of  `claimDeviceOwenrship()` and `isolation.Detect()` is changed
now we call `claimDeviceOwenrship()` before `isolation.Detect()`.
It might affect setting memory limit for libvirtd/QMEU process for VM's with VFIO devices.

File: pkg/virt-handler/vm.go [hunk 1]
```diff
@@ -513,7 +513,8 @@ func (d *VirtualMachineController) setPodNetworkPhase1(vmi *v1.VirtualMachineIns
 	}
 
 	if virtutil.IsNonRootVMI(vmi) && virtutil.WantVirtioNetDevice(vmi) {
-		err := d.claimDeviceOwnership(vmi, "vhost-net")
+		rootMount := res.MountRoot()
+		err := d.claimDeviceOwnership(rootMount, "vhost-net")
```
Comment (ormergi) [inline pkg/virt-handler/vm.go:517]: We need to call `podisolation.Detect()` just before this one, isnt it?

**Comment by EdDev** [comment]
change: Rebase

**Review by alonSadan — APPROVED** [review]

**Comment by alonSadan** [comment]
Thank Eddy

**Comment by EdDev** [comment]
change: Rebase

**Review by enp0s3 — APPROVED** [review]
@EdDev Thank you!
/approve

**Pull request merged**